components/tidb_query_vec_executors/src/table_scan_executor.rs

```diff
@@ -291,21 +295,38 @@ impl ScanExecutorImpl for TableScanExecutorImpl {
     ) -> Result<()> {
         use tidb_query_datatype::codec::{datum, table};
 
-        check_record_key(&key)?;
+        table::check_record_key(&key)?;
         let columns_len = self.schema.len();
         let mut decoded_columns = 0;
 
+        // NOTE: `primary_column_ids` is mutual exclusive toward `pk_handle`.
+        // If the column contains a pk_handle flag, we should extract the value of the column from the key.
         if !self.handle_indices.is_empty() {
```

Comment:
Please document that in this case we expect int handle.

Reply:
OK